app/src/main/java/com/example/tkey_android/EthereumSignerError.java

```diff
@@ -0,0 +1,26 @@
+package com.example.tkey_android;
+
+public class EthereumSignerError extends Error {
+    private ErrorType errorType;
+
+    public EthereumSignerError(ErrorType errorType) {
+        this.errorType = errorType;
+    }
+
+    public String getErrorDescription() {
+        switch (errorType) {
```

Comment:
Probably should add an Insufficient Funds description here, since signing can be successful however the transaction can still fail due to lack of funds